src/Tizen.Applications.ComponentBased/Tizen.Applications.ComponentBased.Common/ComponentBasedApplication.cs

```diff
@@ -0,0 +1,201 @@
+
+            foreach (var component in typeInfo)
+            {
+                RegisterComponent(component.Key, component.Value);
+            }
+        }
+
+        /// <summary>
+        /// Registers a component.
+        /// </summary>
+        /// <param name="compType">Class type</param>
+        /// <param name="compId">Component ID</param>
+        /// <exception cref="ArgumentException">Thrown when component type is already added or not sub-class of FrameComponent or ServiceComponent</exception>
+        /// <since_tizen> 6 </since_tizen>
+        public void RegisterComponent(Type compType, string compId)
+        {
+            if (_componentFactories.ContainsKey(compType))
+            {
+                throw new ArgumentException("Already exist type");
+            }
+
+            if (typeof(FrameComponent).IsAssignableFrom(compType))
+            {
+                Log.Info(LogTag, "Add frame component");
+                _componentFactories.Add(compType, new FrameComponentStateManager(compType, compId, this));
+            }
+            else if (typeof(ServiceComponent).IsAssignableFrom(compType))
+            {
+                Log.Info(LogTag, "Add service component");
+                _componentFactories.Add(compType, new ServiceComponentStateManager(compType, compId, this));
+            }
+            else
+            {
+                throw new ArgumentException("compType must be sub type of FrameComponent or ServiceComponent", "compType");
+            }
+        }
+
+        /// <summary>
+        /// Runs the application's main loop.
+        /// </summary>
+        /// <param name="args">Arguments from commandline.</param>
+        /// <exception cref="InvalidOperationException">Thrown when component type is already added to the component.</exception>
+        /// <since_tizen> 6 </since_tizen>
+        public override void Run(string[] args)
+        {
+            base.Run(args);
+
+            string[] argsClone = new string[args.Length + 1];
+            if (args.Length > 1)
+            {
+                args.CopyTo(argsClone, 1);
+            }
+            argsClone[0] = string.Empty;
+
+            Interop.CBApplication.ErrorCode err = Interop.CBApplication.BaseMain(argsClone.Length, argsClone, ref _callbacks, IntPtr.Zero);
+            if (err != Interop.CBApplication.ErrorCode.None)
+            {
+                Log.Error(LogTag, "Failed to run the application. Err = " + err);
+                throw new InvalidOperationException("Fail to run application : err(" + err + ")");
+            }
+        }
+
+        /// <summary>
+        /// Exits the main loop of the application.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public override void Exit()
+        {
+            Interop.CBApplication.BaseExit();
+        }
+
+        private IntPtr OnCreateNative(IntPtr data)
+        {
+            Log.Debug(LogTag, "On create");
+            IntPtr h = IntPtr.Zero;
```

Comment:
I hope to use a readable and understandable variable naming
```suggestion
            IntPtr nativeComponentFactoryMap = IntPtr.Zero;
```

Reply:
Done